unity-client/Assets/Scripts/MainScripts/DCL/Helpers/NFTHelper/NFTMarkets/OpenSea.cs

```diff
@@ -1,33 +1,41 @@
 ﻿using System;
 using System.Collections;
+using System.Collections.Generic;
 using System.Linq;
-using UnityEngine.Networking;
+using DCL.Helpers.NFT.Markets.OpenSea_Internal;
 
 namespace DCL.Helpers.NFT.Markets
 {
-    public class OpenSea : INFTMarket
+    internal class OpenSea : INFTMarket
     {
-        const string API_URL_SINGLE_ASSET = "https://api.opensea.io/api/v1/asset/{0}/{1}/";
-
         MarketInfo openSeaMarketInfo = new MarketInfo() { name = "OpenSea" };
+        RequestController requestController = new RequestController();
+        Dictionary<string, NFTInfo> cachedResponses = new Dictionary<string, NFTInfo>();
 
         IEnumerator INFTMarket.FetchNFTInfo(string assetContractAddress, string tokenId, Action<NFTInfo> onSuccess, Action<string> onError)
         {
-            string url = string.Format(API_URL_SINGLE_ASSET, assetContractAddress, tokenId);
-            using (UnityWebRequest request = UnityWebRequest.Get(url))
+            string nftId = $"{assetContractAddress}/{tokenId}";
+            if (cachedResponses.ContainsKey(nftId))
             {
-                yield return request.SendWebRequest();
+                onSuccess?.Invoke(cachedResponses[nftId]);
+                yield break;
+            }
 
-                if (!request.isNetworkError && !request.isHttpError)
+            Request request = requestController.AddRequest(assetContractAddress, tokenId);
+            yield return request.OnResolved(
+                (assetResponse) =>
                 {
-                    AssetResponse response = Utils.FromJsonWithNulls<AssetResponse>(request.downloadHandler.text);
-                    onSuccess?.Invoke(ResponseToNFTInfo(response));
-                }
-                else
+                    NFTInfo nftInfo = ResponseToNFTInfo(assetResponse);
+                    onSuccess?.Invoke(nftInfo);
```

Comment:
Why don't you cache the response before resuming invocation flow?

Reply:
> Why don't you cache the response before resuming invocation flow?

You are right, I will change it to cache the response first